optax/contrib/_ademamix.py

```diff
@@ -0,0 +1,220 @@
+"""AdEMAMix.
+
+Implementation of
+"THE ADEMAMIX OPTIMIZER: BETTER, FASTER, OLDER"
+(https://arxiv.org/pdf/2409.03137) by Matteo Pagliardini, 
+Pierre Ablin and David Grangier.
+"""
+
+from typing import NamedTuple, Tuple
+import chex
+import jax.numpy as jnp
+import jax.tree_util as jtu
+from optax._src import base
+from optax._src import combine
+from optax._src import numerics
+from optax._src import transform
+import optax.tree_utils as otu
+
+class ScaleByAdemamixState(NamedTuple):
+  """State for the Ademamix algorithm."""
+
+  count: chex.Array  # shape=(), dtype=jnp.int32.
+  count_m2: chex.Array  # shape=(), dtype=jnp.int32.
+  m1: base.Updates
+  m2: base.Updates
+  nu: base.Updates
+
+
+def scale_by_ademamix(
+  b1: float,
+  b2: float,
+  b3: base.ScalarOrSchedule,
+  alpha: base.ScalarOrSchedule,
+  eps: float = 1e-8,
+  eps_root: float = 0.0,
+) -> base.GradientTransformation:
+  """Rescale updates according to the Ademamix algorithm.
```

Comment:
Could you update this docstring too? The `ademamix` docstring looks great. You may simply refer here to the docstring of `ademamix` for e.g. the pseudocode and copy-paste e.g. the descriptions of the arguments.

Reply:
So to clarify:  just a link to the `ademamix` docstring?  That is, don't follow the convention of:
```
  Description

  References:

  Args:

  Returns:
    A `GradientTransformation` object.
```
?

Note:  the above convention is prevalent throughout `optax/_src/transform.py` for similar `scale_by_X` functions.